library/core/src/num/mod.rs

```diff
@@ -329,6 +329,33 @@ impl u8 {
         *self | (self.is_ascii_uppercase() as u8 * ASCII_CASE_MASK)
     }
 
+    /// Returns the ASCII digit formed by the u8.
+    ///
+    /// # Errors
+    /// Returns `None` if the `char` does not refer to a digit in the given radix.
+    ///
+    /// # Panics
+    /// Panics if given a radix larger than 36.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(ascii_digit_conversion_on_u8)]
+    /// let one = b'1';
+    /// let fifteen = b'F';
+    ///
+    /// assert_eq!(one.to_ascii_digit(10), Some(1));
+    /// assert_eq!(fifteen.to_ascii_digit(16), Some(15));
+    /// ```
+    #[unstable(
+        feature = "ascii_digit_conversion_on_u8",
+        reason = "Newly introduced.",
+        issue = "95969"
+    )]
+    pub fn to_ascii_digit(&self, radix: u8) -> Option<u8> {
```

Comment:
const fn ?

Reply:
> const fn ?

Can't have const closures yet, sorry :(

Reply:
Just don't use closure so ^^

Reply:
> Just don't use closure so ^^

Well.. yeah :P

I could approach this two ways
- Make another method and pass that
- There's some const method that does what the closure does but I'm not aware of it

Reply:
> I could approach this two ways [...]

You can also pattern-match:

```rust
    pub const fn to_ascii_digit(&self, radix: u8) -> Option<u8> {
        if let Some(n) = (*self as char).to_digit(radix.into()) {
            Some(n as u8)
        } else {
            None
        }
    }
```

Note that you probably want this method to take `self` rather than `&self`, like `to_digit()` does.

Reply:
> Note that you probably want this method to take `self` rather than `&self`, like `to_digit()` does
Yah I did think of that, but most of u8 methods take &self and then dereference it under the hood so I just picked up the pattern and did the same 

Changing 👍🏻